enhancements/cert-manager-proxy/http01_challenge_proxy.md

```diff
@@ -0,0 +1,107 @@
+---
+title: http01-challenge-cert-manager-proxy
+authors:
+  - "@sebrandon1"
+reviewers:
+  - "@TrilokGeer"
+  - "@swagosh"
+approvers:
+  - "@tkashem"
+  - "@deads2k"
+  - "@derekwaynecarr"
+api-approvers:
+  - "@JoelSpeed"
+creation-date: 2025-03-28
+last-updated: 2025-03-28
+status: implementable
+tracking-link:
+  - https://issues.redhat.com/browse/CNF-13731
+---
+
+# HTTP01 Challenge Proxy for Cert Manager
+
+![HTTP01 Challenge Proxy Diagram](http01_challenge.png)
+
+## Summary
+
+For baremetal platforms only.  Provide a way for cert-manager to manage certifications for API endpoints (such as api.cluster.example.com) similar to the way it handles certificates for other OpenShift Ingress endpoints.
+
+## Motivation
+
+Cert manager can be used to issue certificates for the OpenShift Container Platform (OCP) endpoints (e.g., console, downloads, oauth) using an external ACME Certificate Authority (CA). These endpoints are exposed via the OpenShift Ingress (`*.apps.cluster.example.com`), and this is a supported and functional configuration today.
+
+However, cluster administrators often want to issue custom certificates for the API endpoint (`api.cluster.example.com`). Unlike other endpoints, this API endpoint is not exposed via the OpenShift Ingress. Depending on the OCP topology (e.g., SNO, MNO, Compact), it is exposed directly on the node or via a keepalive VIP. This lack of management by the OpenShift Ingress introduces challenges in obtaining certificates using an external ACME CA.
+
+The gap arises due to how the ACME HTTP01 challenge works. The following scenarios illustrate the challenges:
+
+1. **SNO (Single Node OpenShift)**: The same nodes host both the ingress and API components. Both FQDNs (`api` and wildcard) resolve to the same IP, making the challenge feasible.
+2. **Compact Clusters**: The node hosting the API VIP may also host an OpenShift Router. If no router is present on the node hosting the VIP, the challenge will fail.
+
+To address this gap, a small proxy was developed. This proxy runs on the cluster as a DaemonSet and ensures that connections reaching the API on port 80 are redirected to the OpenShift Ingress Routers. The proxy implementation uses `nftables` to redirect traffic from `API:80` to `PROXY:8888`.
+
+- **Proxy Code**: [GitHub Repository](https://github.com/mvazquezc/cert-mgr-http01-proxy/tree/main)
+- **Deployment Manifest**: [Manifest Link](https://github.com/mvazquezc/cert-mgr-http01-proxy/blob/main/manifests/deploy-in-ocp.yaml)
+
+This enhancement aims to provide a robust solution for managing certificates for the API endpoint in baremetal environments.
+
+### User Stories
+
+1. **As a cluster administrator**, I want to issue custom certificates for the API endpoint (`api.cluster.example.com`) using an external ACME CA, so that I can ensure secure communication for my cluster's API.
```

Comment:
```suggestion
1. **As a cluster administrator**, I want to manage custom certificates for the API endpoint (`api.cluster.example.com`) using an external ACME CA, so that I can ensure secure communication for my cluster's API.
```